app/models/manageiq/providers/amazon/cloud_manager/metrics_capture.rb

```diff
@@ -131,7 +131,9 @@ def perf_capture_data_amazon(start_time, end_time)
 
     counter_values_by_ts = {}
     COUNTER_INFO.each do |i|
-      timestamps = i[:amazon_counters].collect { |c| metrics_by_counter_name[c].keys }.flatten.uniq.sort
+      timestamps = i[:amazon_counters].collect do |c|
+        metrics_by_counter_name[c].keys unless metrics_by_counter_name[c].nil?
+      end.flatten.uniq.sort
```

Comment:
Shouldn't there be a compact in here now otherwise you will have nils in the timestamps Array.